lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java

```diff
@@ -960,33 +970,75 @@ private static List<Object> decodeRecordReaderStruct(Type type, List<?> fields)
         return newFields;
     }
 
-    public static DataSize writeOrcColumnHive(File outputFile, Format format, CompressionKind compression, Type type, Iterator<?> values)
+    public static void writeOrcColumnHive(File outputFile, Format format, CompressionKind compression, Type type, Iterator<?> values)
             throws Exception
     {
         RecordWriter recordWriter = createOrcRecordWriter(outputFile, format, compression, type);
-        return writeOrcFileColumnHive(outputFile, recordWriter, type, values);
+        writeOrcColumnHive(recordWriter, type, values);
     }
 
-    public static DataSize writeOrcFileColumnHive(File outputFile, RecordWriter recordWriter, Type type, Iterator<?> values)
+    public static void writeOrcColumnHive(RecordWriter recordWriter, Type type, Iterator<?> values)
             throws Exception
     {
-        SettableStructObjectInspector objectInspector = createSettableStructObjectInspector("test", type);
+        StandardStructObjectInspector objectInspector = getStandardStructObjectInspector(ImmutableList.of("test"), ImmutableList.of(getJavaObjectInspector(type)));
+        writeOrcColumnsHive(
+                recordWriter,
+                objectInspector,
+                ImmutableList.of(type),
+                StreamSupport.stream(
+                        Spliterators.spliteratorUnknownSize(values, Spliterator.ORDERED), false)
+                        .map(value -> (Function<Integer, Object>) (fieldIndex) -> value)
+                        .iterator());
+    }
+
+    public static void writeOrcColumnsHiveFile(
+            File outputFile,
+            OrcTester.Format format,
+            CompressionKind compression,
+            List<String> names,
+            List<Type> types,
+            Iterator<Function<Integer, Object>> values)
+            throws Exception
+    {
+        StandardStructObjectInspector objectInspector = getStandardStructObjectInspector(names,
+                types.stream().map(OrcTester::getJavaObjectInspector).collect(toImmutableList()));
+
+        writeOrcColumnsHive(
+                createOrcRecordWriter(outputFile, format, compression, objectInspector),
+                objectInspector,
+                types,
+                values);
+    }
+
+    public static void writeOrcColumnsHive(
+            RecordWriter recordWriter,
+            StandardStructObjectInspector objectInspector,
+            List<Type> types,
+            Iterator<Function<Integer, Object>> values)
+            throws Exception
+    {
+        requireNonNull(types, "types is null");
+        requireNonNull(objectInspector, "objectInspector is null");
+
         Object row = objectInspector.create();
 
         List<StructField> fields = ImmutableList.copyOf(objectInspector.getAllStructFieldRefs());
+        int fieldCount = fields.size();
+        checkArgument(fieldCount != types.size(), "Field and type count must be the same");
```

Comment:
```suggestion
        checkArgument(fieldCount == types.size(), "Field and type count must be the same");
```

Reply:
thank you for finding - my bad, missed it :( . fixed.